indexdata.go

```diff
@@ -376,6 +382,28 @@ func (r *ngramIterationResults) candidates() []*candidateMatch {
 func (d *indexData) iterateNgrams(query *query.Substring) (*ngramIterationResults, error) {
 	str := query.Pattern
 
+	if d.bloomNames.Len() > 0 && len(query.Pattern) >= bloomHashMinWordLength {
+		pat := []byte(query.Pattern)
+		fnmatch := d.bloomNames.maybeHasBytes(pat)
+		if query.FileName {
+			if !fnmatch {
+				return &ngramIterationResults{
+					matchIterator: &noMatchTree{
+						Why: "bloomfilter",
+					},
+				}, nil
+			}
+		} else {
+			if !fnmatch && !d.bloomContents.maybeHasBytes(pat) {
```

Comment:
you aren't checking if bloomContents is non-empty. I realise if name is non-empty this is non-empty, but that feels like a fragile coupling for future changes.

Reply:
I made bloom.maybeHas / bloom.maybeHasBytes always return true when empty to simplify this logic